h3/src/error.rs

```diff
@@ -15,17 +19,38 @@ pub struct Error {
 #[derive(PartialEq, Eq, Hash, Clone, Copy)]
 pub struct Code(u64);
 
+impl Code {
+    pub fn value(&self) -> u64 {
+        self.0
+    }
+}
+
+impl PartialEq<u64> for Code {
+    fn eq(&self, other: &u64) -> bool {
+        *other == self.0
+    }
+}
+
+#[derive(Clone)]
 struct ErrorImpl {
     kind: Kind,
-    cause: Option<Cause>,
+    cause: Option<Arc<Cause>>,
 }
 
-enum Kind {
+#[derive(Clone, Debug)]
+pub enum Kind {
```

Comment:
I agree. I made it public so I can test error cases.

I guess marking everything `#[non_exhaustive]` would not be enough ?

Reply:
I made all the fields and the enum itself `#[non_exheustive]`, hidden it from the docs, and feature-gated the `Error::kind()` accessor behind `test_helpers`: b6479b9

Reply:
If only it were easy to toggle `pub` with an attribute... Oh well, maybe just a comment above that this type is not to be used, it's only exposed for tests, and can change with any release. I only suggest that since sometimes `doc(hidden)` doesn't mean "don't use me", but means "too noisy for docs". Up to you.

Reply:
> If only it were easy to toggle pub with an attribute...
Yeah, looks like we're not the only ones that could find it useful: https://internals.rust-lang.org/t/pub-test-visibility-modifier/12587/4

I added this warning comment. Thanks for the insights.